### PR TITLE
Adding containers to simulate multiple clients

### PR DESCRIPTION
**Pull request by Unic-X**
Some issues that has been fixed: 
- [x] Server can now find the id_rsa
- [x] Clients do have their own TTY (use`docker attach <container-name>` to run those TTY sessions)
- [X] ~~Need of docker networking for client and master to talk to each other~~

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,9 @@ BINARY=goback
 
 all: build
 
+poner: 	#install but in spanish
```
Comment (Unic-X) [inline Makefile:5]: Isko hata deta hu

**Comment by Unic-X** [comment]
@bsach64 bhai wo kuch internal error aa raha mujhe smjh nahi aa raha ye krde bhai jaa raha gym i shall never see you again
image

**Comment by Unic-X** [comment]
For some reason wd + "/private/id_rsa" does not work in docker but does work locally

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+FROM golang:1.23-alpine AS builder
```
Comment (bsach64) [inline Dockerfile:1]: use this simpler dockerfile allows for easier testing

```
FROM debian:stable-slim

WORKDIR /app

COPY goback /app/goback
COPY test_files/ /app/test_files/

RUN apt-get update
RUN apt upgrade -y
RUN apt-get install -y openssh-client

RUN mkdir /app/private
RUN mkdir /app/.data

RUN ssh-keygen -t rsa -b 4096 -f /app/private/id_rsa -N ""

CMD ["goback"]
```

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+services:
```
Comment (bsach64) [inline docker-compose.yml:1]:
```
services:
  server:
    build: .
    command: ["./goback", "server"]
    environment:
      - KEY_PATH=/app/private/id_rsa
    volumes:
      - ./data:/app/data
      - ./private/id_rsa:/app/private/id_rsa
      - ./test_files/:/app/test_files
  client:
    tty: true
    stdin_open: true
    build: .
    command: ["./goback", "client"]
    depends_on:
      - server
    environment:
      - KEY_PATH=/app/private/id_rsa
      - TERM=xterm-256color   
    volumes:
      - ./data:/app/data
      - ./test_files/:/app/test_files
    deploy:
      replicas: 3 
```

slight changes to work with new dockerfile

**Review by bsach64 — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (Unic-X) [reply Dockerfile:1]: Don't use debian Alpine is for faster and smaller container builds

Comment (Unic-X) [reply Dockerfile:1]: Make directory using single command also this won't register changes whenever code is changed bc how Docker handles caching.

Comment (bsach64) [reply Dockerfile:1]: > Make directory using single command also this won't register changes whenever code is changed bc how Docker handles caching.

??

Comment (bsach64) [reply Dockerfile:1]: > Don't use debian Alpine is for faster and smaller container builds

doesn't really matter, we can use alphine
this is mainly for testing right?

Comment (Unic-X) [reply Dockerfile:1]: Receiver ki same IP hai dono me?

Comment (bsach64) [reply Dockerfile:1]: image

image


nhi alag alag hai

Comment (bsach64) [reply Dockerfile:1]: pata nhi
samajh nhi aa rha

**Comment by Unic-X** [comment]
Merge krra abhi to sab chal rah a

**Pull request merged**